[Windows] OS Version minor changes

basharast opened the pull request:
This related to old PR #17914 

When I added UWP system info I used [GetVersionEx](https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getversionexw) 

and I noticed later that this API has very odd note:

> [GetVersionEx may be altered or unavailable for releases after Windows 8.1]

In anyway it's preferred for UWP to use it's own stuff such as `AnalyticsInfo::VersionInfo->DeviceFamilyVersion` 

I added the `OS Build` preview to `System Information` 


## Xbox

For XBox it doesn't make sense to show `Windows 11`, it's better to appear as `XBox OS` with build number below

![image](https://github.com/user-attachments/assets/453d771e-c59e-446a-b10e-dc71ced89536)

Regarding to anr2me comment below it's correct, I was confused too when I searched for this, 
but later I double checked and found the [latest update note](https://support.xbox.com/en-US/help/hardware-network/settings-updates/whats-new-xbox-one-system-updates) which look like they currently call it `XBox OS` (Typical MS).

## UWP Internal

I pulled system info out of `OSVersion.cpp` for UWP, to avoid adding extra params

and UWP has few cases only (10, 11, XBOX), not worth to corrupt `OSVersion.cpp` structure for it

so system info on UWP will be fetched within it's own UWP helpers.


## Desktop

I also added the `OS Build` for the Desktop version hope it's fine.

Let me if there are any comments, Thanks.

basharast committed:
OS Version preview

basharast committed:
OS Version UWP, Win32

basharast committed:
Adding SYSPROP_SYSTEMBUILD

Using `SYSPROP_SYSTEMVERSION` is bad idea since it's set under `Available as Int`, instead I just added `SYSPROP_SYSTEMBUILD`

anr2me commented:
From [wikipedia](https://en.wikipedia.org/wiki/Xbox_system_software):
> According to Polygon, Microsoft removed the distinction between Xbox One and Windows PC,[[39]](https://en.wikipedia.org/wiki/Xbox_system_software#cite_note-xboxpc-39) and the software has been called "Windows 10 on Xbox One".[[41]](https://en.wikipedia.org/wiki/Xbox_system_software#cite_note-41)[[42]](https://en.wikipedia.org/wiki/Xbox_system_software#cite_note-42)

basharast committed:
Simplify & win11-build check fix

Without build check on Windows 10 it will appear as Windows 11, I forgot to keep build check, also the process can be simplified more to avoid extra functions.
